src/main/scala/pipedsl/Parser.scala

```diff
@@ -429,9 +476,20 @@ class Parser extends RegexParsers with PackratParsers {
     "circuit" ~> braces(cseq)
   }
 
+
   lazy val prog: P[Prog] = positioned {
     fdef.* ~ moddef.* ~ circuit ^^ {
       case f ~ p ~ c => Prog(f, p, c)
     }
   }
+
+
+  def parseCode(code :String) :Prog = {
```

Comment:
We should probably move the failure reporting into Main so that the caller can decide what to do with the failure.
So this function should just return the result of parseAll

Reply:
for some reason importing Success/Error/Failure isn't working properly. There might be something I'm missing though